### ISSUE TITLE
Post of subtype not including cast

### ISSUE THREAD

**Issue by phatcher** [object/Simple.OData.Client#306]
I have some code along the lines of 

```
var dog = new Dog { Name = "Fido" };
var result = client.For<Animal>
                   .As<Dog>
                   .Set(dog)
                   .InsertEntryAsync();
```
What gets submitted is 

`POST http://localhost/odata/Animal`

whereas I would expect

`POST http://localhost/odata/Animal/Dog`

Gets includes the typecast, whereas the Post does not; there's specific handling server-side for each type and at the moment because of the missing cast it's arriving at the base type method rather than the specific method for the subtype.

**Comment by object** [comment]
Yes this sounds like a bug. I am quite busy with other project at the moment, can't promise to resolve it promptly but should be able to find some time next week.

**Comment by phatcher** [comment]
Thank Vagif, I've worked around it for now - you seem to have to as otherwise you are reliant on the client being well-behaved and applying the cast.

I'd appreciate an answer to #302 though.

**Comment by habex-ch** [comment]
Any news on this issue?
